Update UPE settings block for BNLP

KarlisJ opened the pull request:
Fixes #6276 

Update the UPE settings card to reflect Affirm and Afterpay as payment methods for those merchant who have the feature available.

The new design is tailored to also account for non-eligible merchants by only removing the icons from the asset.

The new design on desktop
![image](https://github.com/Automattic/woocommerce-payments/assets/24650518/76c69e5c-b3b1-4fdd-9127-29fb495acdbb)

The new design on Mobile
![image](https://github.com/Automattic/woocommerce-payments/assets/24650518/92407a52-61d9-43b7-8648-b17179445cf6)


#### Testing instructions

Hardcode the return value of `WC_Payments_Features::is_bnpl_affirm_afterpay_enabled()` method to both `true` and `false` and observe:
- if feature is not enabled, we should see the asset without "Affirm" and "Afterpay" icons
- if feature is enabled, we should see the asset with "Affirm" and "Afterpay" icons.

-------------------

- [ x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ x] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

Commit:
Add feature flag for BNLP Affirm Afterpay payment method

Commit:
Changelog entry

Commit:
Add unit tests

KarlisJ committed:
Fix a typo

Co-authored-by: Miguel Gasca <miguel.gasca@automattic.com>

Commit:
Corrected changelog

Commit:
Update UPE settings block for BNLP enabled merchants

Commit:
Added changelog

KarlisJ commented:
I made a mistake and based this PR off and branch that's not yet in `develop`. Please review only last 2 commits. 

The merge conflicts should be resolved as soon as [Add feature flag for BNLP Affirm Afterpay payment method](https://github.com/Automattic/woocommerce-payments/pull/6288#top) is merged

nagpai commented:
Manual testing done and it works perfect ✅ 

With BNPL disabled 
![image](https://github.com/Automattic/woocommerce-payments/assets/4162931/8bca4ed4-1bda-4f1d-af3c-2b1c19f4a0d6)


With BNPL enabled
![image](https://github.com/Automattic/woocommerce-payments/assets/4162931/03917b75-920b-4313-825b-23e0c05996d4)

nagpai commented on changelog/add-6034-feature-flag-bnlp:
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: dev
+Comment: This merge is a feature flog utilities that will be made available for masses only in later stages of the project
```
Typo in `flag` .

nagpai replied:
Is this a changelog of a different PR?

KarlisJ replied:
Yes, there's a glitch in GH. It's showing diff to files that are already in `develop`.

nagpai reviewed (approved):
Minor comment on changelog. Preapproving subject to resolving conflicts.

mgascam commented on client/payment-methods/index.js:
```diff
@@ -77,9 +77,41 @@ const UpeSetupBanner = () => {
-	return (
-		<>
-			<CardDivider />
+	let card;
```
I'd suggest a comment explaining that we will remove the condition after the project rollout.

KarlisJ replied:
Not relavant anymore after [this commit ](https://github.com/Automattic/woocommerce-payments/pull/6386/commits/36d24bfe031810fdd904a2fdea28c48004d61fad)

mgascam commented on client/payment-methods/index.js:
Noting that we usually migrate to Typescript when we modify a js file.

KarlisJ replied:
I plan to that, right now focusing on the implementation details

mgascam commented on client/payment-methods/index.js:
Do you think we should add new tests for this file in `client/payment-methods/test/index.js`?

KarlisJ replied:
Yes. Currently we don't have any tests for this component but I'd prefer if we at least created a test for showing correct asset based on feature flag.

That way, once the feature flag will be removed, the test will fail and we'll have to update it to reflect the conditional render on the eligibility of merchant.

Commit:
Unify design between eligible and non-eligible merchants

Through discussion with design team we realised that we can actually reuse the new design for all merchants and only conditionally remove BNPL icons from asset.

KarlisJ commented:
@nagpai given [this commit](https://github.com/Automattic/woocommerce-payments/pull/6386/commits/36d24bfe031810fdd904a2fdea28c48004d61fad) I've updated testing instructions as we no longer support the old design.

Commit:
Update asset alignment

Commit:
Add unit tests

mgascam commented on client/payment-methods/style.scss:
```diff
@@ -60,6 +60,40 @@
+			padding-top: 11px;
+		}
+
+		@media ( max-width: 1024px ) {
```
I noticed that the illustration is not hidden in narrow screens when the `.background-local-payment-methods` is active:

<img width="383" alt="Screenshot 2023-06-05 at 11 30 03" src="https://github.com/Automattic/woocommerce-payments/assets/1553182/c041398b-b71e-49db-afea-fdf2be0800ba">

Can you have a look please?

KarlisJ replied:
Good catch, thank you!

KarlisJ replied:
Addressed [here](https://github.com/Automattic/woocommerce-payments/pull/6386/commits/76639cff1128d48c4316f7ad67c3fd7458df97b0).

Ready for final review.

mgascam reviewed (changes requested):
Finished a manual testing round. I only found a small CSS issue: the illustration overlaps the text in narrow screens when the non-BNPL illustration is displayed. It hides it as expected in BNPL scenario.

Other than that, code looks great. I like the way you manage to reuse the illustration for BNPL and non-BNPL scenarios.

Commit:
Remove background asset for non-eligible mechants

mgascam reviewed (approved):
LGTM!

Commit:
Resolve merge conflicts

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6386` or branch name `update/6276-upe-settings-for-bnlp` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=update/6276-upe-settings-for-bnlp) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 500a3ab13ed2f66c932ef3133356eeb49e6c5482
- Build time: 2023-06-08 05:30:58 UTC

_Note: the build is updated when a new commit is pushed to this PR._

Commit:
Update changelog entry to reflect recent changes

nagpai commented:
Tested with new instructions 

BNPL available:
![image](https://github.com/Automattic/woocommerce-payments/assets/4162931/f9b39899-9143-464f-8fd0-a2948519d374)

BNP not available - banner seems to appear with zero padding 

![image](https://github.com/Automattic/woocommerce-payments/assets/4162931/295992c5-b60b-4932-9e5d-63f6722726da)

Seems to be happening because of this code 

https://github.com/Automattic/woocommerce-payments/blob/15940179275ea19019805a004941bb8538fe8e71/client/payment-methods/style.scss#L5-L6

Something along the lines of this rule will help 

```css
.components-card__body.components-card-body.wcpay-card-body.payment-methods__upe-methods.background-local-payment-methods{
		padding: 24px;
	}
```

KarlisJ commented:
Thanks @nagpai for checking this. I can reproduce the issue but it's not related to BNPL eligibility. The design broke after merging latest changes in the `develop` branch. 

I'll try to compare what's different and from that find the exact cause.

nagpai commented:
It looks like this line that got merged in this [conflict resolve commit](https://github.com/Automattic/woocommerce-payments/commit/44d4147bde0e91de36deedd0559108953c37359f). 

https://github.com/Automattic/woocommerce-payments/blame/44d4147bde0e91de36deedd0559108953c37359f/client/payment-methods/style.scss#L4

Commit:
Use existing class name

Originally we renamed the wrapper class because the whole block was behind feature flag but now we see that exisitng class name is referenced in many other places so decide to leave it untouched

KarlisJ commented:
@nagpai @mgascam Latest commit addressed the issue. Ready for final review.

KarlisJ committed:
Merge branch 'develop' into update/6276-upe-settings-for-bnlp

mgascam reviewed (approved):
After latest change, the illustration shows with the correct padding

jbordonado commented on client/payment-methods/index.js:
```diff
@@ -97,7 +101,7 @@ const UpeSetupBanner = () => {
 				<div className="payment-methods__express-checkouts-actions">
 					<span className="payment-methods__express-checkouts-get-started">
-						<Button isPrimary onClick={ handleEnableUpeClick }>
+						<Button isSecondary onClick={ handleEnableUpeClick }>
```
`isPrimary` and `isSecondary` are both deprecated attributes from the `<Button>` component

Next time, could you try to migrate to the new suggested way (`variant="secondary"` in this case) when you encounter deprecated usages, please?
